### PR TITLE
Import user.group changes from ECS

### PR DESCRIPTION
**Pull request by webmat**
The dependency on https://github.com/elastic/ecs/pull/308 is resolved (see also https://github.com/elastic/ecs/pull/304 for full context). The change is merged. This ECS change lets use nest both the group name and id at `user.group.id` and `user.group.name`.

This PR imports these changes in Beats, and will unblock #9963 and #10192.

**Mapping Conflict Warning**

If you have any index template from Beats 7.0 pre-release in your Elasticsearch, this change introduces a mapping conflict.

Before:

```
{ "user": { "group": "root" } }
```

After this PR:

```
{ "user": { "group": { "id": 0, "name": "root" } } }
```

You will have to overwrite your index templates, delete you indices and your Kibana index templates.

### CODE REVIEW

**Comment by webmat** [comment]
`make check` fails for legit reason: this PR tweaks some ECS fields, and make check detects they haven't made it to x-pack/auditbeat.

However running `make update` in `x-pack/auditbeat/` or `auditbeat/` doesn't update anything more than when I run it at the top of the Beats repo.

@cwurm or @andrewkroh what's the proper incantation, to update field definitions for x-pack/auditbeat?

File: auditbeat/auditbeat.reference.yml [hunk 1]
```diff
@@ -29,60 +29,26 @@ auditbeat.max_start_delay: 10s
 #==========================  Modules configuration =============================
 auditbeat.modules:
 
-# The auditd module collects events from the audit framework in the Linux
```
Comment (andrewkroh) [inline auditbeat/auditbeat.reference.yml]: Did you have DEV_OS=darwin set when you updated? Looks like some thing that should be changed have been changed.

Comment (webmat) [reply auditbeat/auditbeat.reference.yml]: It was `GOOS` and `GOARCH`, I think. I just resubmitted after generating without those, and it got rid of the config file weirdness.

**Comment by webmat** [comment]
Figured out the `make update` problem mentioned above. First legit test run just started, so we need to wait for that.

But otherwise, this is now ready for official review.

**Comment by webmat** [comment]
@andrewkroh Part of this (or separate PR) would surely be to import the generated Go files?

Not sure how/where to integrate those. Can you show me how? Is it applicable here, or is it not yet being used in Beats?

**Comment by andrewkroh** [comment]
We should try to always keep the fields yml and the generated code in sync. It should just be a matter of running

`govendor fetch github.com/elastic/ecs/code/go/ecs` from withing elastic/beats/.

Assuming that you have govendor installed.

**Comment by webmat** [comment]
Jenkins failure limited to Metricbeat, and appears to be due to network flakiness (tar segfault when getting a Docker image, seems like)

**Comment by webmat** [comment]
This is ready for final review. I would like to get this in Thursday, to unblock #9963 and #10192.

@andrewkroh Gotcha, thanks for confirming!

**Comment by webmat** [comment]
Just noticed I hadn't entered a changelog for a change that affects all beats and causes a mapping conflict, LOL

Added the changelog now, and will wait until tomorrow to merge :-)

**Comment by ruflin** [comment]
jenkins, test this

**Comment by webmat** [comment]
@cwurm This has been merged 😃

**Comment by cwurm** [comment]
Thanks a lot @webmat, I‘ve been eagerly awaiting it! :)

**Comment by ruflin** [comment]
Not 100% sure but it seems this PR broke CI. Not sure why this was merged with a red build?

**Comment by ruflin** [comment]
Seems like it was caused by a change in ES: https://github.com/elastic/beats/pull/10336 We should still if possible not merge red PR's.

**Pull request merged**